bottom-nav-watson/src/main/kotlin/bottom_nav_watson/BottomNavWatson.kt

```diff
@@ -36,96 +32,90 @@ fun BottomNavigationView.setupWithNavController(
 
     // When a navigation item is selected
     setOnNavigationItemSelectedListener { item ->
         // Don't do anything if the state is state has already been saved.
         if (fragmentManager.isStateSaved) {
             false
         } else {
-            val newlySelectedItemTag = graphIdToTagMap[item.itemId]
+            var newlySelectedItemTag = tabIdToTagMap[item.itemId]
+            if (newlySelectedItemTag == null) {
+                initNavController(
+                    activity = activity,
+                    index = enabledTabs.indexOf(item.itemId),
+                    tabId = item.itemId,
+                    graphResId = graphResId,
+                    containerId = containerId,
+                    tabIdToTagMap = tabIdToTagMap,
+                    selectedNavController = selectedNavController,
+                    destinationChangedListener = destinationChangedListener
+                )
+
+                newlySelectedItemTag = tabIdToTagMap[item.itemId]
+            }
+
             if (selectedItemTag != newlySelectedItemTag) {
-                // Pop everything above the first fragment (the "fixed start destination")
+                // Pop everything above the initial fragment (the "fixed start destination")
                 fragmentManager.popBackStack(
-                    firstFragmentTag,
+                    initialFragmentTag,
                     FragmentManager.POP_BACK_STACK_INCLUSIVE
                 )
+
                 val selectedFragment = fragmentManager.findFragmentByTag(newlySelectedItemTag)
                         as NavHostFragment
 
-                // Exclude the first fragment tag because it's always in the back stack.
-                if (firstFragmentTag != newlySelectedItemTag) {
-                    // Commit a transaction that cleans the back stack and adds the first fragment
-                    // to it, creating the fixed started destination.
-                    fragmentManager.beginTransaction()
-                        .attach(selectedFragment)
-                        .setPrimaryNavigationFragment(selectedFragment)
-                        .apply {
-                            // Detach all other Fragments
-                            graphIdToTagMap.forEach { _, fragmentTagIter ->
-                                if (fragmentTagIter != newlySelectedItemTag) {
-                                    detach(fragmentManager.findFragmentByTag(firstFragmentTag)!!)
-                                }
+                // Commit a transaction that cleans the back stack and adds the initial fragment
+                // to it, creating the fixed started destination.
+                fragmentManager.beginTransaction()
+                    .attach(selectedFragment)
+                    .setPrimaryNavigationFragment(selectedFragment)
+                    .apply {
+                        // Detach all other Fragments
+                        tabIdToTagMap.forEach { _, fragmentTagIter ->
```

Comment:
[nit][naming] what does the `Iter` suffix mean? I'd leave it out :grimacing:

Reply:
Copy/pasted from the original Google sources, I'll rename it 👍 But I guess it scopes the naming to the iterable item of the forEach loop

Reply:
> Nice 🚀 It even seems to have simplified the code nicely :)

Yes, now I can completely understand the code xD